--- a/src/popthelp.c
+++ b/src/popthelp.c
@@ -15,7 +15,7 @@
 #include <sys/ioctl.h>
 #endif

-#ifdef HAVE_MBSRTOWCS
+#if 0 /* IRIX */
 #include <wchar.h>			/* for mbsrtowcs */
 #endif
 #include "poptint.h"
@@ -118,7 +118,7 @@ static inline size_t displayLength(const char * s)
 static inline size_t stringDisplayWidth(const char *s)
 {
     size_t n = strlen(s);
-#ifdef HAVE_MBSRTOWCS
+#if 0 /* IRIX */
     mbstate_t t;

     memset ((void *)&t, 0, sizeof (t));	/* In initial state.  */
