diff -u libsecret-0.21.3/egg/egg-secure-memory.c.orig libsecret-0.21.3/egg/egg-secure-memory.c
--- libsecret-0.21.3/egg/egg-secure-memory.c.orig
+++ libsecret-0.21.3/egg/egg-secure-memory.c
@@ -28,6 +28,13 @@

 #include "config.h"

+#if defined(__sgi)
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#endif
+
 #include "egg-secure-memory.h"

 #include <sys/types.h>
@@ -166,6 +173,9 @@
 {
 	Pool *pool;
 	void *pages, *item;
+#if defined(__sgi)
+	int mmapfd;
+#endif
 	size_t len, i;

 	if (!EGG_SECURE_GLOBALS.pool_version ||
@@ -188,7 +198,15 @@
 	/* Create a new pool */
 	if (pool == NULL) {
 		len = getpagesize () * 2;
+#if defined(__sgi)
+		if ((mmapfd = open("/dev/zero", O_RDWR)) == -1) {
+			return NULL;
+		}
+		pages = mmap (0, len, PROT_READ | PROT_WRITE, MAP_PRIVATE, mmapfd, 0);
+		close(mmapfd);
+#else
 		pages = mmap (0, len, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
+#endif
 		if (pages == MAP_FAILED)
 			return NULL;

@@ -852,6 +870,9 @@
 {
 	void *pages;
 	unsigned long pgsize;
+#if defined(__sgi)
+	int mmapfd;
+#endif

 	ASSERT (sz);
 	ASSERT (*sz);
@@ -863,7 +884,15 @@
 	*sz = (*sz + pgsize -1) & ~(pgsize - 1);

 #if defined(HAVE_MLOCK)
+#if defined(__sgi)
+	mmapfd = open("/dev/zero", O_RDWR);
+	if (mmapfd == -1)
+		return NULL;
+	pages = mmap (0, *sz, PROT_READ | PROT_WRITE, MAP_PRIVATE, mmapfd, 0);
+	close(mmapfd);
+#else
 	pages = mmap (0, *sz, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
+#endif
 	if (pages == MAP_FAILED) {
 		if (show_warning && egg_secure_warnings)
 			fprintf (stderr, "couldn't map %lu bytes of memory (%s): %s\n",
