diff -ruN glib-2.80.0/gio/glocalfile.c glib-2.80.0/gio/glocalfile.c
--- glib-2.80.0/gio/glocalfile.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/gio/glocalfile.c	2026-02-10 03:14:35.486688943 +0000
@@ -2721,7 +2721,7 @@
       if (!error)
         return FALSE;
 
-#ifdef AT_FDCWD
+#if defined(AT_FDCWD) && !defined(__sgi)
       /* If using openat() we need to rebuild the filename for the message */
       filename = g_string_new (name->data);
       for (node = name->next; node; node = node->next)
@@ -2915,14 +2915,14 @@
   if (S_ISDIR (_g_stat_mode (&buf)))
     {
       int dir_fd = -1;
-#ifdef AT_FDCWD
+#if defined(AT_FDCWD) && !defined(__sgi)
       int errsv;
 #endif
 
       if (g_cancellable_set_error_if_cancelled (state->cancellable, error))
         return FALSE;
 
-#ifdef AT_FDCWD
+#if defined(AT_FDCWD) && !defined(__sgi)
 #ifdef HAVE_OPEN_O_DIRECTORY
       dir_fd = openat (parent_fd, name->data, O_RDONLY | O_DIRECTORY | O_CLOEXEC);
 #else
@@ -2951,7 +2951,7 @@
   GDir *dir;
   gint saved_errno;
 
-#ifdef AT_FDCWD
+#if defined(AT_FDCWD) && !defined(__sgi)
   {
     /* If this fails, we want to preserve the errno from fdopendir() */
     DIR *dirp;
@@ -2967,7 +2967,7 @@
 
   if (dir == NULL)
     {
-#ifdef AT_FDCWD
+#if defined(AT_FDCWD) && !defined(__sgi)
       close (fd);
 #endif
 
@@ -2979,7 +2979,7 @@
       GSList node;
 
       node.next = dir_name;
-#ifdef AT_FDCWD
+#if defined(AT_FDCWD) && !defined(__sgi)
       node.data = (gchar *) name;
 #else
       node.data = g_build_filename (dir_name->data, name, NULL);
@@ -3018,7 +3018,7 @@
   state.progress_callback = progress_callback;
   state.progress_data = progress_data;
 
-#ifdef AT_FDCWD
+#if defined(AT_FDCWD) && !defined(__sgi)
   root_fd = AT_FDCWD;
 #endif
 
diff -ruN glib-2.80.0/gio/gunixfdmessage.c glib-2.80.0/gio/gunixfdmessage.c
--- glib-2.80.0/gio/gunixfdmessage.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/gio/gunixfdmessage.c	2026-02-10 02:53:29.481863274 +0000
@@ -39,6 +39,10 @@
 #include <fcntl.h>
 #include <errno.h>
 
+#if defined(__sgi)
+# define SCM_RIGHTS 0x01
+#endif
+
 #include "gunixfdmessage.h"
 #include "gunixfdlist.h"
 #include "gnetworking.h"
diff -ruN glib-2.80.0/gio/gunixmounts.c glib-2.80.0/gio/gunixmounts.c
--- glib-2.80.0/gio/gunixmounts.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/gio/gunixmounts.c	2026-02-10 02:53:29.482863289 +0000
@@ -82,6 +82,7 @@
 #include "gthemedicon.h"
 #include "gcontextspecificgroup.h"
 
+static const char * DEVROOTPATH = "/dev/otherroot";
 
 #ifdef HAVE_MNTENT_H
 static const char *_resolve_dev_root (void);
@@ -523,7 +524,7 @@
       gboolean is_read_only = FALSE;
 
       device_path = mnt_fs_get_source (fs);
-      if (g_strcmp0 (device_path, "/dev/root") == 0)
+      if (g_strcmp0 (device_path, DEVROOTPATH) == 0)
         device_path = _resolve_dev_root ();
 
       mount_options = mnt_fs_strdup_options (fs);
@@ -571,6 +572,51 @@
 G_LOCK_DEFINE_STATIC(getmntent);
 #endif
 
+/* Helper struct + function to create a g_slist containing the
+ * mntent entries so we don't have to G_LOCK
+ */
+typedef struct mntent_entry
+{
+  struct mntent mntent;
+} mntent_entry_s;
+
+static GSList * get_gslist_getmntent( FILE * file )
+{
+  GSList * ret_gslist=NULL;
+  GSList * el=NULL;
+  struct mntent *mntent;
+
+  G_LOCK (getmntent);
+  while ((mntent = getmntent (file)) != NULL)
+    {
+      mntent_entry_s * val = malloc(sizeof(mntent_entry_s));
+      val->mntent = *mntent;
+      el = g_slist_append( el, val );
+    }
+  G_UNLOCK (getmntent);
+  if( el != NULL ) {
+    ret_gslist = el;
+  }
+
+  return ret_gslist;
+}
+
+static void free_gslist_getmntent( GSList * gslist )
+{
+  if( gslist != NULL ) {
+    GSList * el = gslist;
+    do
+      {
+        void * data = el->data;
+        mntent_entry_s * val = (mntent_entry_s *)data;
+        free(val);
+      }
+    while( (el = g_slist_next(el)) != NULL );
+
+    g_slist_free( gslist );
+  }
+}
+
 static GList *
 _g_get_unix_mounts (void)
 {
@@ -584,7 +630,12 @@
   GUnixMountEntry *mount_entry;
   GHashTable *mounts_hash;
   GList *return_list;
+  GSList * mntent_list;
+  GSList * ent;
   
+  const char *device_path;
+  gboolean is_read_only;
+
   read_file = get_mtab_read_file ();
 
   file = setmntent (read_file, "re");
@@ -597,13 +648,20 @@
   
 #ifdef HAVE_GETMNTENT_R
   while ((mntent = getmntent_r (file, &ent, buf, sizeof (buf))) != NULL)
+    {
 #else
-  G_LOCK (getmntent);
-  while ((mntent = getmntent (file)) != NULL)
+    // G_LOCK (getmntent);
+    // while ((mntent = getmntent (file)) != NULL)
+    mntent_list = get_gslist_getmntent( file );
+  if( mntent_list != NULL ) {
+    ent = mntent_list;
+    do {
+      void * data = ent->data;
+      mntent_entry_s * val = (mntent_entry_s *)data;
+      mntent = &(val->mntent);
 #endif
-    {
-      const char *device_path = NULL;
-      gboolean is_read_only = FALSE;
+      device_path = NULL;
+      is_read_only = FALSE;
 
       /* ignore any mnt_fsname that is repeated and begins with a '/'
        *
@@ -620,7 +678,7 @@
 	  g_hash_table_lookup (mounts_hash, mntent->mnt_fsname))
         continue;
 
-      if (g_strcmp0 (mntent->mnt_fsname, "/dev/root") == 0)
+      if (g_strcmp0 (mntent->mnt_fsname, DEVROOTPATH) == 0)
         device_path = _resolve_dev_root ();
       else
         device_path = mntent->mnt_fsname;
@@ -642,7 +700,13 @@
 			   mount_entry->device_path);
 
       return_list = g_list_prepend (return_list, mount_entry);
+#if defined(HAVE_GETMNTENT_R)
     }
+#else
+  } while( (ent = g_slist_next(ent)) != NULL );
+    }
+  free_gslist_getmntent( mntent_list );
+#endif
   g_hash_table_destroy (mounts_hash);
   
   endmntent (file);
@@ -1076,7 +1140,7 @@
         }
 
       device_path = mnt_fs_get_source (fs);
-      if (g_strcmp0 (device_path, "/dev/root") == 0)
+      if (g_strcmp0 (device_path, DEVROOTPATH) == 0)
         device_path = _resolve_dev_root ();
 
       mount_point = create_unix_mount_point (device_path,
@@ -1145,7 +1209,7 @@
         continue;
 #endif
 
-      if (strcmp (mntent->mnt_fsname, "/dev/root") == 0)
+      if (strcmp (mntent->mnt_fsname, DEVROOTPATH) == 0)
         device_path = _resolve_dev_root ();
       else
         device_path = mntent->mnt_fsname;
@@ -3229,7 +3293,7 @@
   /* otherwise we're going to find it right away.. */
   have_real_dev_root = TRUE;
   
-  if (stat ("/dev/root", &statbuf) == 0) 
+  if (stat (DEVROOTPATH, &statbuf) == 0) 
     {
       if (! S_ISLNK (statbuf.st_mode)) 
         {
@@ -3276,7 +3340,7 @@
        else 
         {
           char *resolved;
-          resolved = _resolve_symlink ("/dev/root");
+          resolved = _resolve_symlink (DEVROOTPATH);
           if (resolved != NULL)
             {
               strncpy (real_dev_root, resolved, sizeof (real_dev_root) - 1);
@@ -3288,7 +3352,7 @@
     }
   
   /* bah sucks.. */
-  strcpy (real_dev_root, "/dev/root");
+  strcpy (real_dev_root, DEVROOTPATH);
   
 found:
   return real_dev_root;
diff -ruN glib-2.80.0/gio/tests/desktop-app-info.c glib-2.80.0/gio/tests/desktop-app-info.c
--- glib-2.80.0/gio/tests/desktop-app-info.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/gio/tests/desktop-app-info.c	2026-02-10 02:53:29.483863303 +0000
@@ -155,9 +155,9 @@
 
   g_object_unref (info);
 
-  if (g_file_test ("/usr/share/applications/gedit.desktop", G_FILE_TEST_EXISTS))
+  if (g_file_test ("/usr/sgug/share/applications/gedit.desktop", G_FILE_TEST_EXISTS))
     {
-      info = (GAppInfo*)g_desktop_app_info_new_from_filename ("/usr/share/applications/gedit.desktop");
+      info = (GAppInfo*)g_desktop_app_info_new_from_filename ("/usr/sgug/share/applications/gedit.desktop");
       g_assert_nonnull (info);
      
       res = g_app_info_can_delete (info);
diff -ruN glib-2.80.0/gio/tests/socket.c glib-2.80.0/gio/tests/socket.c
--- glib-2.80.0/gio/tests/socket.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/gio/tests/socket.c	2026-02-10 02:53:29.485863332 +0000
@@ -2558,11 +2558,14 @@
                        &error);
   if (sock != NULL)
     {
+      printf("Assuming ipv6 ok\n");
       ipv6_supported = TRUE;
       g_object_unref (sock);
     }
   else
     {
+      printf("Assuming ipv6 not ok\n");
+      errno = 0;
       g_assert_error (error, G_IO_ERROR, G_IO_ERROR_NOT_SUPPORTED);
       g_clear_error (&error);
     }
diff -ruN glib-2.80.0/gio/xdgmime/xdgmime.c glib-2.80.0/gio/xdgmime/xdgmime.c
--- glib-2.80.0/gio/xdgmime/xdgmime.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/gio/xdgmime/xdgmime.c	2026-02-10 02:53:29.486863347 +0000
@@ -228,7 +228,7 @@
   xdg_data_dirs = getenv ("XDG_DATA_DIRS");
 
   if (xdg_data_dirs == NULL)
-    xdg_data_dirs = "/usr/local/share/:/usr/share/";
+    xdg_data_dirs = "/usr/sgug/share/:/usr/share/";
 
   /* Work out how many dirs we’re dealing with. */
   if (xdg_data_home != NULL || home != NULL)
diff -ruN glib-2.80.0/glib/gcharset.c glib-2.80.0/glib/gcharset.c
--- glib-2.80.0/glib/gcharset.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/gcharset.c	2026-02-10 02:55:58.952058617 +0000
@@ -503,7 +503,7 @@
   if (g_once_init_enter_pointer (&alias_table))
     {
       GHashTable *table = g_hash_table_new (g_str_hash, g_str_equal);
-      read_aliases ("/usr/share/locale/locale.alias", table);
+      read_aliases ("/usr/sgug/share/locale/locale.alias", table);
       g_once_init_leave_pointer (&alias_table, table);
     }
 
diff -ruN glib-2.80.0/glib/giounix.c glib-2.80.0/glib/giounix.c
--- glib-2.80.0/glib/giounix.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/giounix.c	2026-02-10 02:53:29.487863362 +0000
@@ -33,7 +33,17 @@
 
 #include "config.h"
 
+#if defined(__sgi)
+/* Needed to get the right declarations for select() before POSIX_SOURCE
+ * define
+ */
+#include <unistd.h>
+#include <sys/types.h>
+#include <bstring.h>
+#include <sys/time.h>
+#else
 #define _POSIX_SOURCE		/* for SSIZE_MAX */
+#endif
 
 #include <sys/types.h>
 #include <sys/stat.h>
diff -ruN glib-2.80.0/glib/glibconfig.h.in glib-2.80.0/glib/glibconfig.h.in
--- glib-2.80.0/glib/glibconfig.h.in	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/glibconfig.h.in	2026-02-10 02:53:29.487863362 +0000
@@ -24,7 +24,7 @@
  * system printf functions.  This is useful to know, for example,
  * when using glibc's register_printf_function().
  */
-#mesondefine GLIB_USING_SYSTEM_PRINTF
+/*#mesondefine GLIB_USING_SYSTEM_PRINTF*/
 
 G_BEGIN_DECLS
 
diff -ruN glib-2.80.0/glib/gmain.c glib-2.80.0/glib/gmain.c
--- glib-2.80.0/glib/gmain.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/gmain.c	2026-02-10 02:53:29.488863377 +0000
@@ -2955,7 +2955,11 @@
   struct timespec ts;
   gint result;
 
+#if defined(__sgi)
+  result = clock_gettime (CLOCK_SGI_CYCLE, &ts);
+#else
   result = clock_gettime (CLOCK_MONOTONIC, &ts);
+#endif
 
   if G_UNLIKELY (result != 0)
     g_error ("GLib requires working CLOCK_MONOTONIC");
diff -ruN glib-2.80.0/glib/gmessages.c glib-2.80.0/glib/gmessages.c
--- glib-2.80.0/glib/gmessages.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/gmessages.c	2026-02-10 02:53:29.489863391 +0000
@@ -920,9 +920,11 @@
     }
   else
     {
-      GError *err = NULL;
+      GError *err;
+      gchar *result;
+      err = NULL;
+      result = g_convert_with_fallback (string, -1, charset, "UTF-8", "?", NULL, NULL, &err);
       
-      gchar *result = g_convert_with_fallback (string, -1, charset, "UTF-8", "?", NULL, NULL, &err);
       if (result)
 	return result;
       else
diff -ruN glib-2.80.0/glib/gnulib/fpucw.h glib-2.80.0/glib/gnulib/fpucw.h
--- glib-2.80.0/glib/gnulib/fpucw.h	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/gnulib/fpucw.h	2026-02-10 02:53:29.489863391 +0000
@@ -88,6 +88,36 @@
 # define END_LONG_DOUBLE_ROUNDING() \
   SET_FPUCW (oldcw)
 
+#elif defined(__sgi)
+
+typedef unsigned int fpucw_t;
+
+# include <sys/fpu.h>
+
+# define FPU_PC_MASK 0
+# define FPU_PC_DOUBLE 0
+# define FPU_PC_EXTENDED 0
+
+# define GET_FPUCW() 0
+# define SET_FPUCW(word) (void)(word)
+
+# define DECL_LONG_DOUBLE_ROUNDING              \
+  union fpc_csr old_csr, new_csr;
+
+# define BEGIN_LONG_DOUBLE_ROUNDING()                     \
+  do {                                                    \
+    old_csr.fc_word = get_fpc_csr();                      \
+    new_csr.fc_word = old_csr.fc_word;                    \
+    new_csr.fc_struct.flush=0;                            \
+    new_csr.fc_struct.rounding_mode=ROUND_TO_NEAREST;     \
+    set_fpc_csr(new_csr.fc_word);                         \
+  } while (0)
+
+# define END_LONG_DOUBLE_ROUNDING()             \
+  do {                                          \
+    set_fpc_csr(old_csr.fc_word);               \
+  } while (0)
+
 #else
 
 typedef unsigned int fpucw_t;
diff -ruN glib-2.80.0/glib/gnulib/gl_cv_func_frexp_works/meson.build glib-2.80.0/glib/gnulib/gl_cv_func_frexp_works/meson.build
--- glib-2.80.0/glib/gnulib/gl_cv_func_frexp_works/meson.build	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/gnulib/gl_cv_func_frexp_works/meson.build	2026-02-10 03:08:08.856926222 +0000
@@ -91,10 +91,13 @@
     gl_cv_func_frexp_broken_beyond_repair = false
   endif
 else
-  if (host_system.startswith ('netbsd') or
-      host_system.startswith ('irix'))
+  if host_system.startswith ('netbsd')
     gl_cv_func_frexp_works = false
     gl_cv_func_frexp_broken_beyond_repair = true
+  elif host_system.startswith ('irix')
+    # IRIX 6.5 libm frexp works correctly
+    gl_cv_func_frexp_works = true
+    gl_cv_func_frexp_broken_beyond_repair = false
   elif host_system == 'windows'
     if cc.get_id () == 'msvc' or cc.get_id() == 'clang-cl'
       gl_cv_func_frexp_works = true
diff -ruN glib-2.80.0/glib/gnulib/gl_cv_func_frexpl_works/meson.build glib-2.80.0/glib/gnulib/gl_cv_func_frexpl_works/meson.build
--- glib-2.80.0/glib/gnulib/gl_cv_func_frexpl_works/meson.build	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/gnulib/gl_cv_func_frexpl_works/meson.build	2026-02-10 03:08:13.971002806 +0000
@@ -120,10 +120,13 @@
   endif
 else
   if (host_system.startswith ('aix') or
-      host_system.startswith ('beos') or
-      host_system.startswith ('irix'))
+      host_system.startswith ('beos'))
     gl_cv_func_frexpl_works = false
     gl_cv_func_frexpl_broken_beyond_repair = true
+  elif host_system.startswith ('irix')
+    # IRIX 6.5 libm frexpl works correctly
+    gl_cv_func_frexpl_works = true
+    gl_cv_func_frexpl_broken_beyond_repair = false
   elif (host_system == 'windows')
     gl_cv_func_frexpl_works = false
     gl_cv_func_frexpl_broken_beyond_repair = false
diff -ruN glib-2.80.0/glib/gnulib/vasnprintf.c glib-2.80.0/glib/gnulib/vasnprintf.c
--- glib-2.80.0/glib/gnulib/vasnprintf.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/gnulib/vasnprintf.c	2026-02-10 02:53:29.491863421 +0000
@@ -3072,7 +3072,7 @@
                         int sign = 0;
                         DECL_LONG_DOUBLE_ROUNDING
 
-                        BEGIN_LONG_DOUBLE_ROUNDING ();
+                        BEGIN_LONG_DOUBLE_ROUNDING();
 
                         if (signbit (arg)) /* arg < 0.0L or negative zero */
                           {
diff -ruN glib-2.80.0/glib/goption.c glib-2.80.0/glib/goption.c
--- glib-2.80.0/glib/goption.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/goption.c	2026-02-10 02:53:29.492863435 +0000
@@ -713,7 +713,11 @@
   if (context->help_enabled ||
       (context->main_group && context->main_group->n_entries > 0) ||
       context->groups != NULL)
+#if defined(__sgi)
+    g_string_append_printf (string, " %s", _("[OPTION.]"));
+#else
     g_string_append_printf (string, " %s", _("[OPTION…]"));
+#endif
 
   if (rest_description)
     {
diff -ruN glib-2.80.0/glib/gthread-posix.c glib-2.80.0/glib/gthread-posix.c
--- glib-2.80.0/glib/gthread-posix.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/gthread-posix.c	2026-02-10 02:53:29.493863450 +0000
@@ -678,7 +678,7 @@
 
   pthread_condattr_init (&attr);
 
-#ifdef HAVE_PTHREAD_COND_TIMEDWAIT_RELATIVE_NP
+#if defined(HAVE_PTHREAD_COND_TIMEDWAIT_RELATIVE_NP) || defined(__sgi)
 #elif defined (HAVE_PTHREAD_CONDATTR_SETCLOCK) && defined (CLOCK_MONOTONIC)
   if G_UNLIKELY ((status = pthread_condattr_setclock (&attr, CLOCK_MONOTONIC)) != 0)
     g_thread_abort (status, "pthread_condattr_setclock");
@@ -918,6 +918,22 @@
     if ((status = pthread_cond_timedwait_relative_np (g_cond_get_impl (cond), g_mutex_get_impl (mutex), &ts)) == 0)
       return TRUE;
   }
+#elif defined(__sgi)
+  {
+    gint64 now = g_get_monotonic_time();
+    gint64 absendtime;
+
+    if (end_time <= now)
+      return FALSE;
+
+    absendtime = g_get_real_time () + (end_time - now);
+
+    ts.tv_sec = absendtime / 1000000;
+    ts.tv_nsec = (absendtime % 1000000) * 1000;
+
+    if ((status = pthread_cond_timedwait (g_cond_get_impl (cond), g_mutex_get_impl (mutex), &ts)) == 0)
+      return TRUE;
+  }
 #elif defined (HAVE_PTHREAD_CONDATTR_SETCLOCK) && defined (CLOCK_MONOTONIC)
   /* This is the exact check we used during init to set the clock to
    * monotonic, so if we're in this branch, timedwait() will already be
diff -ruN glib-2.80.0/glib/gtimezone.c glib-2.80.0/glib/gtimezone.c
--- glib-2.80.0/glib/gtimezone.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/gtimezone.c	2026-02-10 02:55:57.901043182 +0000
@@ -511,13 +511,15 @@
 static const gchar *
 zone_info_base_dir (void)
 {
-  if (g_file_test ("/usr/share/zoneinfo", G_FILE_TEST_IS_DIR))
+  if (g_file_test ("/usr/sgug/share/zoneinfo", G_FILE_TEST_IS_DIR))
+    return "/usr/sgug/share/zoneinfo"; /* IRIX/SGUG */
+  else if (g_file_test ("/usr/share/zoneinfo", G_FILE_TEST_IS_DIR))
     return "/usr/share/zoneinfo";     /* Most distros */
   else if (g_file_test ("/usr/share/lib/zoneinfo", G_FILE_TEST_IS_DIR))
     return "/usr/share/lib/zoneinfo"; /* Illumos distros */
 
   /* need a better fallback case */
-  return "/usr/share/zoneinfo";
+  return "/usr/sgug/share/zoneinfo";
 }
 
 static gchar *
diff -ruN glib-2.80.0/glib/gutils.c glib-2.80.0/glib/gutils.c
--- glib-2.80.0/glib/gutils.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/gutils.c	2026-02-10 02:55:55.197003472 +0000
@@ -1077,11 +1077,17 @@
       {
         glong max;
 
+#if defined(__sgi)
+        max = 0;
+#else
         max = sysconf (_SC_HOST_NAME_MAX);
+#endif
         if (max > 0 && (gsize) max <= G_MAXSIZE - 1)
           size = (gsize) max + 1;
         else
-#ifdef HOST_NAME_MAX
+#if defined(__sgi)
+          size = MAXHOSTNAMELEN + 1;
+#elif defined(HOST_NAME_MAX)
           size = HOST_NAME_MAX + 1;
 #else
           size = _POSIX_HOST_NAME_MAX + 1;
@@ -2695,7 +2701,7 @@
    */
 #ifndef G_OS_WIN32
   if (!data_dirs || !data_dirs[0])
-    data_dirs = "/usr/local/share/:/usr/share/";
+    data_dirs = "/usr/sgug/share/:/usr/share/";
 
   data_dir_vector = g_strsplit (data_dirs, G_SEARCHPATH_SEPARATOR_S, 0);
 #else
diff -ruN glib-2.80.0/glib/tests/collate.c glib-2.80.0/glib/tests/collate.c
--- glib-2.80.0/glib/tests/collate.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/tests/collate.c	2026-02-10 02:53:29.496863494 +0000
@@ -105,6 +105,172 @@
   do_collate (TRUE, TRUE, test);
 }
 
+/* No UTF8 locales on IRIX, so qsort won't have expected behaviour
+ * so lets redefine the test sets + results
+ */
+#if defined(__sgi)
+
+const gchar *input0[] = {
+  "z",
+  "c",
+  "eer34",
+  "223",
+  "er1",
+  "foo",
+  "bar",
+  "baz",
+  "GTK+",
+  NULL
+};
+
+const gchar *sorted0[] = {
+  "223",
+  "bar",
+  "baz",
+  "c",
+  "eer34",
+  "er1",
+  "foo",
+  "GTK+",
+  "z",
+  NULL
+};
+
+const gchar *file_sorted0[] = {
+  "223",
+  "bar",
+  "baz",
+  "c",
+  "eer34",
+  "er1",
+  "foo",
+  "GTK+",
+  "z",
+  NULL
+};
+
+const gchar *input1[] = {
+  "file.txt",
+  "file2.bla",
+  "file.c",
+  "file3.xx",
+  "bla001",
+  "bla02",
+  "bla03",
+  "bla4",
+  "bla10",
+  "bla100",
+  "event.c",
+  "eventgenerator.c",
+  "event.h",
+  NULL
+};
+
+const gchar *sorted1[] = {
+  "bla001",
+  "bla02",
+  "bla03",
+  "bla10",
+  "bla100",
+  "bla4",
+  "eventgenerator.c",
+  "event.c",
+  "event.h",
+  "file2.bla",
+  "file3.xx",
+  "file.c",
+  "file.txt",
+  NULL
+};
+
+const gchar *file_sorted1[] = {
+  "bla001",
+  "bla02",
+  "bla03",
+  "bla4",
+  "bla10",
+  "bla100",
+  "event.c",
+  "event.h",
+  "eventgenerator.c",
+  "file.c",
+  "file.txt",
+  "file2.bla",
+  "file3.xx",
+  NULL
+};
+
+const gchar *input2[] = {
+  "file26",
+  "file100",
+  "file1",
+  "file:foo",
+  "a.a",
+  "file027",
+  "file10",
+  "aa.a",
+  "file5",
+  "file0027",
+  "a-.a",
+  "file0000",
+  "file000x",
+  NULL
+};
+
+const gchar *sorted2[] = {
+  "aa.a",
+  "a-.a",
+  "a.a",
+  "file0000",
+  "file000x",
+  "file0027",
+  "file027",
+  "file1",
+  "file10",
+  "file100",
+  "file26",
+  "file5",
+  "file:foo",
+  NULL
+};
+
+const gchar *file_sorted2[] = {
+  /* Filename collation in OS X follows Finder style which gives
+   * a slightly different order from usual Linux locales. */
+#ifdef HAVE_CARBON
+  "a-.a",
+  "a.a",
+  "aa.a",
+  "file:foo",
+  "file0000",
+  "file000x",
+  "file1",
+  "file5",
+  "file10",
+  "file26",
+  "file0027",
+  "file027",
+  "file100",
+#else
+  "a.a",
+  "aa.a",
+  "a-.a",
+  "file0000",
+  "file000x",
+  "file1",
+  "file5",
+  "file10",
+  "file26",
+  "file027",
+  "file0027",
+  "file100",
+  "file:foo",
+#endif
+  NULL
+};
+
+#else
+
 const gchar *input0[] = {
   "z",
   "c",
@@ -267,6 +433,8 @@
   NULL
 };
 
+#endif
+
 int
 main (int argc, char *argv[])
 {
diff -ruN glib-2.80.0/glib/tests/date.c glib-2.80.0/glib/tests/date.c
--- glib-2.80.0/glib/tests/date.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/tests/date.c	2026-02-10 02:53:29.496863494 +0000
@@ -1111,6 +1111,11 @@
 
   g_test_bug ("https://bugzilla.gnome.org/show_bug.cgi?id=793550");
 
+#if defined(__sgi)
+  g_test_skip ("pl_PL locale not available");
+  return;
+#endif
+
   if (setlocale (LC_ALL, "pl_PL") == NULL)
     {
       g_test_skip ("pl_PL locale not available");
diff -ruN glib-2.80.0/glib/tests/gvariant.c glib-2.80.0/glib/tests/gvariant.c
--- glib-2.80.0/glib/tests/gvariant.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/tests/gvariant.c	2026-02-10 02:53:29.499863538 +0000
@@ -19,6 +19,7 @@
 
 #include <glib/gvariant-internal.h>
 #include <string.h>
+#include <stdio.h>
 #include <stdlib.h>
 #include <glib.h>
 
@@ -3906,7 +3907,8 @@
   gchar *pt, *p;
   gchar *res;
 
-  tree = tree_instance_new (NULL, 3);
+  //  tree = tree_instance_new (NULL, 3);
+  tree = tree_instance_new (NULL, 2);
   value = tree_instance_get_gvariant (tree);
   tree_instance_free (tree);
 
@@ -3915,6 +3917,8 @@
 
   parsed = g_variant_parse (NULL, pt, NULL, NULL, NULL);
   res = g_variant_print (parsed, FALSE);
+  //  printf("p=\n%s\n", p);
+  //  printf("res=\n%s\n", res);
   g_assert_cmpstr (p, ==, res);
   g_variant_unref (parsed);
   g_free (res);
diff -ruN glib-2.80.0/glib/tests/protocol.c glib-2.80.0/glib/tests/protocol.c
--- glib-2.80.0/glib/tests/protocol.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/tests/protocol.c	2026-02-10 02:53:29.499863538 +0000
@@ -22,6 +22,8 @@
  * if advised of the possibility of such damage.
  */
 
+#include <stdio.h>
+
 #include <errno.h>  /* errno */
 #include <glib.h>
 #ifdef G_OS_UNIX
diff -ruN glib-2.80.0/glib/tests/strfuncs.c glib-2.80.0/glib/tests/strfuncs.c
--- glib-2.80.0/glib/tests/strfuncs.c	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/glib/tests/strfuncs.c	2026-02-10 02:53:29.500863553 +0000
@@ -1193,7 +1193,11 @@
 {
   gchar *str_end = NULL;
   double value = 0.0;
+#if defined(__sgi)
   const double gold_ratio = 1.61803398874989484;
+#else
+  const double gold_ratio = 1.61803398874989484;
+#endif
   const gchar *gold_ratio_str = "1.61803398874989484";
   const gchar *minus_gold_ratio_str = "-1.61803398874989484";
 
@@ -1707,7 +1711,7 @@
 #endif
   check_strtod_string ("-.75,0", -0.75, TRUE, 4);
 
-#ifndef _MSC_VER
+#if !defined(_MSC_VER) && !defined(__sgi)
   /* the values of d in the following 2 tests generate a C1064 compiler limit error */
   d = 179769313486231570814527423731704356798070567525844996598917476803157260780028538760589558632766878171540458953514382464234321326889464182768467546703537516986049910576551282076245490090389328944075868508455133942304583236903222948165808559332123348274797826204144723168738177180919299881250404026184124858368.0;
   g_assert_true (d == g_ascii_strtod (g_ascii_dtostr (buffer, sizeof (buffer), d), NULL));
diff -ruN glib-2.80.0/gobject/tests/genmarshal.py glib-2.80.0/gobject/tests/genmarshal.py
--- glib-2.80.0/gobject/tests/genmarshal.py	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/gobject/tests/genmarshal.py	2026-02-10 02:53:29.500863553 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/sgug/bin/python3
 # -*- coding: utf-8 -*-
 #
 # Copyright © 2019 Endless Mobile, Inc.
diff -ruN glib-2.80.0/gobject/tests/mkenums.py glib-2.80.0/gobject/tests/mkenums.py
--- glib-2.80.0/gobject/tests/mkenums.py	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/gobject/tests/mkenums.py	2026-02-10 02:53:29.500863553 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/sgug/bin/python3
 # -*- coding: utf-8 -*-
 #
 # Copyright © 2018 Endless Mobile, Inc.
diff -ruN glib-2.80.0/meson.build glib-2.80.0/meson.build
--- glib-2.80.0/meson.build	2024-03-07 21:35:05.000000000 +0000
+++ glib-2.80.0/meson.build	2026-02-10 02:55:51.654951454 +0000
@@ -689,7 +689,6 @@
   'link',
   'localtime_r',
   'lstat',
-  'mbrtowc',
   'memalign',
   'mmap',
   'newlocale',
@@ -716,7 +715,6 @@
   'valloc',
   'vasprintf',
   'vsnprintf',
-  'wcrtomb',
   'wcslen',
   'wcsnlen',
   'sysctlbyname',
