--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,13 +30,14 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)

 # Ensure that we don't accidentally install into /usr/etc
-set(CMAKE_INSTALL_FULL_SYSCONDIR "/etc")
-set(SYSCONFDIR /etc)
+# IRIX SGUG paths
+set(CMAKE_INSTALL_FULL_SYSCONDIR "/usr/sgug/etc")
+set(SYSCONFDIR /usr/sgug/etc)

-set(MOTGEN_DIR /etc/motdgen.d)
+set(MOTGEN_DIR /usr/sgug/etc/motdgen.d)

 if(NOT SYSTEMD_DIR)
-    set(SYSTEMD_DIR /lib/systemd/system)
+    set(SYSTEMD_DIR /usr/sgug/lib/systemd/system)
 endif()

 if(NOT HISTORY_DB_DIR)
