--- a/client/gpgcheck.c
+++ b/client/gpgcheck.c
@@ -430,6 +430,30 @@
 
         for (i = 0; ppszUrlGPGKeys[i]; i++) {
             pr_info("importing key from %s\n", ppszUrlGPGKeys[i]);
+            /* SGUG-RSE: Expand $releasever and $basearch in GPG key URLs */
+            if (strstr(ppszUrlGPGKeys[i], TDNF_VAR_RELEASEVER) ||
+                strstr(ppszUrlGPGKeys[i], TDNF_VAR_BASEARCH)) {
+                char *tempUrl = NULL;
+                PTDNF_CONF pConf = pTdnf->pConf;
+
+                pr_info("found $releasever and $basearch in gpg url, expanding...\n");
+                dwError = TDNFConfigExpandVars(pTdnf);
+                BAIL_ON_TDNF_ERROR(dwError);
+
+                dwError = TDNFReplaceString(
+                            ppszUrlGPGKeys[i],
+                            TDNF_VAR_RELEASEVER,
+                            pConf->pszVarReleaseVer,
+                            &tempUrl);
+                BAIL_ON_TDNF_ERROR(dwError);
+
+                dwError = TDNFReplaceString(
+                            tempUrl,
+                            TDNF_VAR_BASEARCH,
+                            pConf->pszVarBaseArch,
+                            &ppszUrlGPGKeys[i]);
+                BAIL_ON_TDNF_ERROR(dwError);
+                pr_info("importing expanded key from %s\n", ppszUrlGPGKeys[i]);
+            }
+
             dwError = TDNFYesOrNo(pTdnf->pArgs, "Is this ok [y/N]: ", &nAnswer);
             BAIL_ON_TDNF_ERROR(dwError);
 
             if(!nAnswer)
