--- a/solv/tdnfpool.c
+++ b/solv/tdnfpool.c
@@ -109,13 +109,9 @@

     if (pszArch == NULL) {
-        if (uname(&systemInfo)) {
-            dwError = ERROR_TDNF_SOLV_IO;
-            BAIL_ON_TDNF_LIBSOLV_ERROR(dwError);
-        }
-        pool_setarch(pPool, systemInfo.machine);
-    } else {
-        pool_setarch(pPool, pszArch);
+        /* IRIX: Force mips architecture instead of using uname */
+        pszArch = "mips";
     }
+    pool_setarch(pPool, pszArch);

     pool_set_flag(pPool, POOL_FLAG_ADDFILEPROVIDESFILTERED, 1);
