--- a/client/remoterepo.c
+++ b/client/remoterepo.c
@@ -48,9 +48,17 @@
     if (!isatty(STDOUT_FILENO))
     {
+#if defined(__sgi)
+        pr_info("%s %u%% %lld\n", pData->pszData, dPercent, dlNow);
+#else
         pr_info("%s %u%% %ld\n", pData->pszData, dPercent, dlNow);
+#endif
     }
     else
     {
+#if defined(__sgi)
+        pr_info("%-35s %10lld %u%%\r", pData->pszData, dlNow, dPercent);
+#else
         pr_info("%-35s %10ld %u%%\r", pData->pszData, dlNow, dPercent);
+#endif
     }

--- a/tools/cli/lib/api.c
+++ b/tools/cli/lib/api.c
@@ -358,7 +358,11 @@

         if(dwCount > 0)
         {
+#if defined(__sgi)
+            pr_crit("\nTotal Size: %s (%llu)\n", pszFormattedSize, dwTotalSize);
+#else
             pr_crit("\nTotal Size: %s (%lu)\n", pszFormattedSize, dwTotalSize);
+#endif
         }
     }

